docs/api-guide.md

```diff
@@ -0,0 +1,722 @@
+# Java API Guide
+
+This guide briefly explains how to use Scalar DB Java API.
+
+* [Administrative operations](#administrative-operations)
```

Comment:
I just realized that we should probably name them like `Administrative API` and `Transactional API` since it's an API guide?
```suggestion
* [Administrative API](#administrative-api)
```